### PR TITLE
fix: loading of wallet history, display of tx datetime

### PR DESCRIPTION
**Pull request by emizzle**
Fixes: #2539.

Transaction history is now correctly being fetched from status-go as per mobile. Firstly, when accounts are added (ie as watch accounts), `wallet_checkRecentHistory` must be called first so that the status-go db is populated. After that, `wallet_getTransfersByAddress` can be called. On app load, when we run the `initBalance` logic, we are calling `wallet_getTransfersByAddress`, asking for the last 20 transactions with the `loadMore` parameter set to false. When the user navigates to the Wallet > History tab, they can then click “Load More” to fetch more transactions from status-go. Once the number of transactions returns false below the expected amount, the remaining transactions to fetch have been exhausted and the “Load More” button is disabled.

feat: add non-archival node warning to the UI to indicate to the user that they may not have complete results

feat: set active account to the added account
Once an account is added to the wallet, that newly added account is selected as the active account.

### NOTES
1. The “load more” button is active when new transactions that aren’t already displayed are returned from `wallet_getTransfersByAddress`. This is the only way to enable or disable the “Load more” button as status-go is not able to return information regarding whether or not there are more transactions to be fetched. The downside to this is that lets say the last page of transactions is returned, but there are no more pages left. These returned txs are not currently displayed, so the “load more” button will still be enabled. However, the next click of the button will return 0 results, thus disabling it. It’s effectively an extra click to get to the disabled state.
2. For more information on how the `toBlock` parameter operates for the `wallet_getTransfersForAddress` RPC call, see https://notes.status.im/XmENTrVRRaqhwE4gK0m8Mg?view.

### CODE REVIEW

File: src/app/wallet/core.nim [hunk 1]
```diff
@@ -64,13 +64,17 @@ proc init*(self: WalletController) =
...
         self.view.setHistoryFetchState(data.accounts, true)
       of "recent-history-ready":
         self.view.setHistoryFetchState(data.accounts, false)
-        self.view.initBalances(false)
+        self.view.getLatestTransactionHistory(data.accounts)
+        self.view.initBalances()
+      of "non-archival-node-detected":
+        self.view.setHistoryFetchState(self.status.wallet.accounts.map(account => account.address), false)
+        self.view.setNonArchivalNode()
+        error "Non-archival node detected, please check your Infura key or your connected node"
```
Comment (jrainville) [inline src/app/wallet/core.nim:77]: Smart!

File: ui/app/AppLayouts/Wallet/HistoryTab.qml [hunk 2]
```diff
@@ -129,100 +132,107 @@ Item {
...
                     anchors.rightMargin: Style.current.smallPadding
                 }
             }
         }
     }
 
+    StyledText {
+        id: nonArchivalNodeError
+        visible: walletModel.isNonArchivalNode
+        height: visible ? 20 : 0
```
Comment (jrainville) [inline ui/app/AppLayouts/Wallet/HistoryTab.qml:214]: Text handles it's own height  with `implicitHeight`. That way, you don't have to change the `20` if you change the font.
```suggestion
        height: visible ? implicitHeight : 0
```

File: ui/app/AppLayouts/Wallet/HistoryTab.qml [hunk 3]
```diff
@@ -129,100 +132,107 @@ Item {
...
+        text: qsTr("Status Desktop is connected to a non-archival node. Transaction history may be incomplete.")
+        font.pixelSize: Style.current.primaryTextFontSize
+        color: Style.current.danger
+    }
+
+    StyledText {
+        id: noTxs
+        anchors.top: nonArchivalNodeError.bottom
+        visible: transactionListRoot.count === 0
+        height: visible ? 36 : 0
```
Comment (jrainville) [inline ui/app/AppLayouts/Wallet/HistoryTab.qml:225]: I see now that you put different heights to create spacing. If you want, you can leave it as is, but it's probably better to handle the spacing with the anchors margins.
```suggestion
        height: visible ? implicitHeight : 0
```

**Review by jrainville — APPROVED** [review]
Nice job!

**Comment by anastasiyaig** [comment]
@emizzle sorry if i dont get the limitation you described, i have an account with 2500+ txns, and Status app shows only a few, and More button is not active, is it due to the limitations on status-go?  I dont see any errors anywhere, just More is disabled


https://user-images.githubusercontent.com/82375995/120804638-93ef0180-c54d-11eb-8af4-569c2bbdbb2b.mov

**Comment by emizzle** [comment]
@anastasiyaig Can you try deleting your data folder?

**Comment by anastasiyaig** [comment]
> @anastasiyaig Can you try deleting your data folder?

but it is fresh install, imported account

**Comment by emizzle** [comment]
@anastasiyaig  which wallet account is it? Was it imported or added as a watch-only? Maybe DM me?

**Comment by emizzle** [comment]
@anastasiyaig -- ah I see it now, you need to click "ERC-20 Transactions" to see what the data that status-go is giving back.

**Comment by anastasiyaig** [comment]
so the problem is the following: Load More button is not active in the following case:

1. install the PR build
2. import an account with huge history (I can provide it)
3. switch to ropsten (the app must be restarted)
4. open the app again -> enable wallet
5. open history tab

As a result, Load More is not active, however there are a lot of txns for this account https://ropsten.etherscan.io/address/0xc8f8b3376814d0dfe61e4b9eaf7970697c24f6b2

Screenshot 2021-06-07 at 09 40 33

NOTE: additional restart does fix the issue, so if you restart the app -> load more is active.

**Comment by anastasiyaig** [comment]
@emizzle i am not able to find any new issues in the PR, only existing ones that are in master branch
@jrainville @iurimatias could you please review this PR as well , maybe i missed something? I want to merge it but want an additional eyes since there are again status-go changes. To me it works nice and the issue i mentioned before is gone.

**Review by jrainville — APPROVED** [review]

**Pull request merged**